io/src/main/scala/fs2/io/file/file.scala

```diff
@@ -123,6 +127,72 @@ package object file {
       .resource(Watcher.default(blocker))
       .flatMap(w => Stream.eval_(w.watch(path, types, modifiers)) ++ w.events(pollTimeout))
 
+  /**
+    * Checks if a file exists
+    *
+    * Note that the result of this method is immediately outdated. If this
+    * method indicates the file exists then there is no guarantee that a
+    * subsequence access will succeed. Care should be taken when using this
+    * method in security sensitive applications.
+    */
+  def exists[F[_]: Sync](path: Path, flags: Seq[LinkOption] = Seq.empty): F[Boolean] =
+    Sync[F].delay(Files.exists(path, flags: _*))
```

Comment:
Should this use `Blocker` since it's doing FS access?

Reply:
I wasn't sure in cases like these, since in general this should be super quick in most cases, but sure, I'll add it :)

Reply:
Yeah, I've felt the same way. Flip side is that pretty much everything in `Files` API grabs a global mutex when looking up a file system.

Reply:
Very good point 👍

Reply:
I think it's important to use `Blocker`, even here. Think about checking existence on an NFS mount, or even something more benign like a FUSE mount.